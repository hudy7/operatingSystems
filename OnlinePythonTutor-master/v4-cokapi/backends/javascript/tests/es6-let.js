function foo() {
  var xxx = 5;
  if (true) {
    let xxx = 10;
    console.log(xxx);
  }
  console.log(xxx);
}

foo()
