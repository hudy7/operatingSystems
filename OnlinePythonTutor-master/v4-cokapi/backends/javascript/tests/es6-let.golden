{ code: 'function foo() {\n  var xxx = 5;\n  if (true) {\n    let xxx = 10;\n    console.log(xxx);\n  }\n  console.log(xxx);\n}\n\nfoo()',
  trace: 
   [ { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { foo: [ 'REF', 1 ] },
       ordered_globals: [ 'foo' ],
       line: 10,
       col: 0,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'foo',
             'function foo() {\n  var xxx = 5;\n  if (true) {\n    let xxx = 10;\n    console.log(xxx);\n  }\n  console.log(xxx);\n}',
             null,
             null ] } },
     { stdout: '',
       func_name: 'foo',
       stack_to_render: 
        [ { func_name: 'foo',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'foo_f2',
            ordered_varnames: [ 'xxx' ],
            encoded_locals: { xxx: [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { foo: [ 'REF', 1 ] },
       ordered_globals: [ 'foo' ],
       line: 2,
       col: 2,
       event: 'call',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'foo',
             'function foo() {\n  var xxx = 5;\n  if (true) {\n    let xxx = 10;\n    console.log(xxx);\n  }\n  console.log(xxx);\n}',
             null,
             null ] } },
     { stdout: '',
       func_name: 'foo',
       stack_to_render: 
        [ { func_name: 'foo',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'foo_f2',
            ordered_varnames: [ 'xxx' ],
            encoded_locals: { xxx: 5 } } ],
       globals: { foo: [ 'REF', 1 ] },
       ordered_globals: [ 'foo' ],
       line: 3,
       col: 2,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'foo',
             'function foo() {\n  var xxx = 5;\n  if (true) {\n    let xxx = 10;\n    console.log(xxx);\n  }\n  console.log(xxx);\n}',
             null,
             null ] } },
     { stdout: '',
       func_name: 'foo',
       stack_to_render: 
        [ { func_name: 'foo',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'foo_f2',
            ordered_varnames: [ 'xxx' ],
            encoded_locals: { xxx: 5 } } ],
       globals: { foo: [ 'REF', 1 ] },
       ordered_globals: [ 'foo' ],
       line: 4,
       col: 4,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'foo',
             'function foo() {\n  var xxx = 5;\n  if (true) {\n    let xxx = 10;\n    console.log(xxx);\n  }\n  console.log(xxx);\n}',
             null,
             null ] } },
     { stdout: '',
       func_name: 'foo',
       stack_to_render: 
        [ { func_name: 'foo',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'foo_f2',
            ordered_varnames: [ 'xxx' ],
            encoded_locals: { xxx: 5 } } ],
       globals: { foo: [ 'REF', 1 ] },
       ordered_globals: [ 'foo' ],
       line: 5,
       col: 4,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'foo',
             'function foo() {\n  var xxx = 5;\n  if (true) {\n    let xxx = 10;\n    console.log(xxx);\n  }\n  console.log(xxx);\n}',
             null,
             null ] } },
     { stdout: '10\n',
       func_name: 'foo',
       stack_to_render: 
        [ { func_name: 'foo',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'foo_f2',
            ordered_varnames: [ 'xxx' ],
            encoded_locals: { xxx: 5 } } ],
       globals: { foo: [ 'REF', 1 ] },
       ordered_globals: [ 'foo' ],
       line: 7,
       col: 2,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'foo',
             'function foo() {\n  var xxx = 5;\n  if (true) {\n    let xxx = 10;\n    console.log(xxx);\n  }\n  console.log(xxx);\n}',
             null,
             null ] } },
     { stdout: '10\n5\n',
       func_name: 'foo',
       stack_to_render: 
        [ { func_name: 'foo',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'foo_f2',
            ordered_varnames: [ 'xxx', '__return__' ],
            encoded_locals: { xxx: 5, __return__: [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { foo: [ 'REF', 1 ] },
       ordered_globals: [ 'foo' ],
       line: 7,
       col: 0,
       event: 'return',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'foo',
             'function foo() {\n  var xxx = 5;\n  if (true) {\n    let xxx = 10;\n    console.log(xxx);\n  }\n  console.log(xxx);\n}',
             null,
             null ] } },
     { stdout: '10\n5\n',
       func_name: '',
       stack_to_render: [],
       globals: { foo: [ 'REF', 1 ] },
       ordered_globals: [ 'foo' ],
       line: 10,
       col: 5,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'foo',
             'function foo() {\n  var xxx = 5;\n  if (true) {\n    let xxx = 10;\n    console.log(xxx);\n  }\n  console.log(xxx);\n}',
             null,
             null ] } } ] }
